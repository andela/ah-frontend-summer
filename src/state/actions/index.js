/**
 * This module contains all actions for the project
 * They are further split into synchronous (trigger state update immediately)
 * and asynchronous actions (trigger an operation that will ultimately trigger a synchronous action)
 **/
