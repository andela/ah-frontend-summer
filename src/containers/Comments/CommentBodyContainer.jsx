import React, {Component} from "react";
import CKEditor from '@ckeditor/ckeditor5-react';
import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
import {Button} from "semantic-ui-react";
import PropTypes from "prop-types";
import {SUCCEEDED} from "../../store/actions/async";

const commentBoxConfigEdit = {
    toolbar: [
        'bold',
        'italic',
        '|',
        'bulletedList',
        'numberedList',
        'blockQuote',
        '|',
        'undo',
        'redo'
    ],
    placeholder: 'Update comment...'
};

class CommentBodyContainer extends Component {
    state = {
        body: this.props.body
    };

    shouldComponentUpdate(nextProps, nextState, nextContext) {
        return !!this.editor;
    }

    componentDidUpdate(prevProps) {
        const {loading, status, finishEditing} = this.props;
        if (prevProps.loading && !loading) {
            if (status === SUCCEEDED) {
                finishEditing();
            }
        }
    }

    render() {
        const {body} = this.state;
        const {finishEditing, editing, loading} = this.props;
        const bodyIsEmpty = body.length === 0;
        const editorClass = editing ? 'nunito' : 'nunito disabled';
        const buttons = !editing ? '' : (
            <div className="comment-new-comment-buttons">
                <Button
                    disabled={bodyIsEmpty || loading} loading={loading}
                    onClick={() => {
                        this.updateComment();
                    }} color="teal">
                    Save
                </Button>
                <Button
                    negative onClick={() => {
                        this.resetBody();
                        finishEditing();
                    }}>
                    Cancel
                </Button>
            </div>
        );
        return (
            <div className={editorClass}>
                <CKEditor
                    disabled={!editing}
                    editor={ClassicEditor}
                    config={commentBoxConfigEdit}
                    data={body}
                    onChange={(event, editor) => {
                        this.setState({
                            body: editor.getData()
                        });
                    }}
                    onInit={(editor) => {
                        this.editor = editor;
                    }}/>
                {buttons}
            </div>
        );
    }

    updateComment() {
        const {pk, editComment} = this.props;
        const {body} = this.state;
        editComment(pk, body);
    }

    resetBody() {
        const {body} = this.props;
        this.setState({body});
    }
}

CommentBodyContainer.propTypes = {
    body: PropTypes.string.isRequired,
    editComment: PropTypes.func.isRequired,
    editing: PropTypes.bool.isRequired,
    finishEditing: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
    pk: PropTypes.number.isRequired,
    status: PropTypes.string.isRequired
};

export default CommentBodyContainer;
