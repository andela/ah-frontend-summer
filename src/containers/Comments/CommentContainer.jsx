import React, {Component} from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import Comment from "../../components/comments/Comment";
import {
    deleteCommentAction,
    updateCommentAction
} from "../../store/actions/async/CommentActions";
import {getLoggedInUser, SUCCEEDED} from "../../store/actions/async";

class CommentContainer extends Component {
    deleteComment() {
        const {comment, doDeleteComment} = this.props;
        doDeleteComment(comment.id);
    }

    render() {
        const {deleteLoading, editComment} = this.props;
        return (
            <Comment
                loggedInUser={getLoggedInUser(localStorage)}
                editComment={editComment} type="comment"
                deleteLoading={deleteLoading}
                deleteComment={() => this.deleteComment()} {...this.props} />
        )}
}

export const mapDispatchToProps = (dispatch) => {
    return {
        editComment: (pk, body) => dispatch(updateCommentAction(pk, body)),
        doDeleteComment: (id) => dispatch(deleteCommentAction(id))
    };
};

export const mapStateToProps = (state, props) => {
    const pk = props.comment.id;
    const comment = state.comments.comments.filter((comment) => {
        return comment.id === pk;
    })[0];
    return {
        loading: typeof comment === "undefined" ? false : !!comment.loading,
        status: typeof comment === "undefined" ? SUCCEEDED : comment.status ? comment.status : SUCCEEDED,
        deleteLoading: state.comments.deleteComment.loading
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(CommentContainer);

CommentContainer.propTypes = {
    comment: PropTypes.object.isRequired,
    deleteLoading: PropTypes.bool.isRequired,
    doDeleteComment: PropTypes.func.isRequired,
    editComment: PropTypes.func.isRequired
};
