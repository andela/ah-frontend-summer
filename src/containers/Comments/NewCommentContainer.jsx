import React, {Component} from "react";
import CKEditor from '@ckeditor/ckeditor5-react';
import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
import {Button, Segment} from "semantic-ui-react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {createCommentAction} from "../../store/actions/async/CommentActions";
import {SUCCEEDED} from "../../store/actions/async";

export class NewCommentContainer extends Component {
    state = {
        body: "",
    };

    componentDidUpdate(prevProps) {
        const {loading, status} = this.props;
        if (prevProps.loading && !loading) {
            if (status === SUCCEEDED) {
                this.clearBody();
            }
        }
    }

    createNewComment() {
        const {body} = this.state;
        const {slug, createComment} = this.props;
        createComment(slug, body);
    }

    clearBody() {
        this.setState({
            body: ""
        });
    }

    render() {
        const {body} = this.state;
        const {loading, placeholder} = this.props;
        const bodyIsEmpty = body.length === 0;
        const commentBoxConfig = {
            toolbar: [
                'bold',
                'italic',
                '|',
                'bulletedList',
                'numberedList',
                'blockQuote',
                '|',
                'undo',
                'redo'
            ],
            placeholder: placeholder
        };
        return (
            <Segment>
                <CKEditor
                    className="nunito"
                    editor={ClassicEditor}
                    config={commentBoxConfig}
                    data={body}
                    onChange={(event, editor) => {
                        this.setState({
                            body: editor.getData()
                        });
                    }}
                />
                <div className="comment-new-comment-buttons">
                    <Button
                        disabled={bodyIsEmpty || loading} loading={loading}
                        onClick={() => {
                            this.createNewComment();
                        }} color="teal">
                        Save
                    </Button>
                    <Button
                        negative onClick={() => {
                            this.clearBody();
                        }}>
                        Cancel
                    </Button>
                </div>
            </Segment>
        );
    }
}

NewCommentContainer.propTypes = {
    createComment: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
    placeholder: PropTypes.string.isRequired,
    slug: PropTypes.any.isRequired,
    status: PropTypes.string.isRequired
};

export const mapStateToProps = (state) => {
    const newComment = state.comments.newComment;
    const {loading, status} = newComment;
    return {
        loading,
        status
    };
};

export const mapDispatchToProps = (dispatch) => {
    return {
        createComment: (slug, body) => dispatch(createCommentAction(slug, body))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(NewCommentContainer);
