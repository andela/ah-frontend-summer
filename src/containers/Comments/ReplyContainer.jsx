import React, {Component} from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {getLoggedInUser, SUCCEEDED} from "../../store/actions/async";
import {
    deleteReplyAction,
    updateReplyAction
} from "../../store/actions/async/CommentActions";
import Comment from "../../components/comments/Comment";

class ReplyContainer extends Component {
    deleteReply() {
        const {reply, doDeleteReply} = this.props;
        doDeleteReply(reply.id, reply.comment);
    }

    render() {
        const {deleteLoading, reply} = this.props;
        return (
            <Comment
                loggedInUser={getLoggedInUser(localStorage)}
                editComment={(pk, body) => this.editReply(pk, body)}
                type="reply" deleteLoading={deleteLoading}
                deleteComment={() => this.deleteReply()}
                comment={reply} {...this.props} />
        );
    }

    editReply(pk, body) {
        const {doEditReply, reply} = this.props;
        doEditReply(pk, reply.comment, body);
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        doEditReply: (pk, commentPk, body) => dispatch(updateReplyAction(pk, commentPk, body)),
        doDeleteReply: (id, commentPk) => dispatch(deleteReplyAction(id, commentPk))
    };
};

const mapStateToProps = (state, props) => {
    const pk = props.reply.id;
    const commentPk = props.reply.comment;
    const commentRepliesState = state.replies[commentPk];
    let reply = commentRepliesState.replies.filter((reply) => {
        return reply.id === pk;
    })[0];
    return {
        deleteLoading: commentRepliesState.deleteReply ? commentRepliesState.deleteReply.loading : false,
        deleteStatus: commentRepliesState.deleteReply ? commentRepliesState.deleteReply.status : SUCCEEDED,
        loading: reply.loading ? reply.loading : false,
        status: reply.status ? reply.status : SUCCEEDED
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(ReplyContainer);

ReplyContainer.propTypes = {
    deleteLoading: PropTypes.bool.isRequired,
    doDeleteReply: PropTypes.func.isRequired,
    doEditReply: PropTypes.func.isRequired,
    reply: PropTypes.object.isRequired
};
