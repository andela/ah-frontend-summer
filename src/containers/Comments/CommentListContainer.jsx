import React, {Component} from 'react';
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Button, Comment, Header} from "semantic-ui-react";
import {Link} from "react-router-dom";
import CommentList from "../../components/comments/CommentList";
import {
    fetchCommentsAction,
    isLoggedIn
} from "../../store/actions/async/CommentActions";
import CommentLoading from "../../components/comments/CommentLoading";
import {SUCCEEDED} from "../../store/actions/async";
import CommentError from "../../components/comments/CommentError";
import NewCommentContainer from "./NewCommentContainer";


class CommentListContainer extends Component {
    componentDidMount() {
        this.fetchCommentsData();
    }

    fetchCommentsData() {
        const {slug, fetchComments} = this.props;
        fetchComments(slug);
    }

    render() {
        const {comments, loading, status, slug} = this.props;
        let commentsContentToRender;
        if (loading) {
            commentsContentToRender = (
                <CommentLoading />
            );
        } else {
            if (status === SUCCEEDED) {
                commentsContentToRender = <CommentList comments={comments} />;
            } else {
                commentsContentToRender = (
                    <CommentError
                        retry={() => this.fetchCommentsData()}
                        message="Could not load Comments" />
                );
            }
        }
        return (
            <Comment.Group size="large" className="comment-list">
                <Header
                    className="comment-teal" as="h1"
                    dividing>
                    Comments
                </Header>
                {isLoggedIn(localStorage) ? (
                    <NewCommentContainer
                        placeholder="Add a comment..."
                        slug={slug} />
                ): (
                    <div className="center">
                        <Link to="/login">
                            <Button size="large" color="teal">
                                Login to have your say
                            </Button>
                        </Link>
                    </div>
                )}
                {commentsContentToRender}
            </Comment.Group>
        );
    }
}

CommentListContainer.propTypes = {
    loading: PropTypes.bool,
    status: PropTypes.string.isRequired,
    fetchComments: PropTypes.func.isRequired,
    comments: PropTypes.array.isRequired,
    slug: PropTypes.string.isRequired
};

CommentListContainer.defaultProps = {
    loading: true,
};

const mapStateToProps = (state) => {
    return {
        comments: state.comments.comments,
        loading: state.comments.loading,
        status: state.comments.status
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        fetchComments: (slug) => {
            dispatch(fetchCommentsAction(slug));
        }
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(CommentListContainer);
