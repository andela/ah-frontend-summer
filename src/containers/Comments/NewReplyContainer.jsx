import {connect} from "react-redux";
import {createReplyAction} from "../../store/actions/async/CommentActions";
import {NewCommentContainer} from "./NewCommentContainer";
import {SUCCEEDED} from "../../store/actions/async";

const mapStateToProps = (state, props) => {
    const commentPk = props.slug;
    const singleReplyState = state.replies[commentPk];
    let newReply = singleReplyState && singleReplyState.newReply ? singleReplyState.newReply : {
        loading: false,
        status: SUCCEEDED
    };
    return {
        loading: typeof singleReplyState === "undefined" ? false : newReply.loading,
        status: typeof singleReplyState === "undefined" ? SUCCEEDED : newReply.status
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        createComment: (pk, body) => dispatch(createReplyAction(pk, body))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(NewCommentContainer);
