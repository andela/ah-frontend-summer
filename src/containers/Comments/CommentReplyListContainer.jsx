import {Button, Comment, Header} from "semantic-ui-react";
import React, {Component} from "react";
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import ReplyList from "../../components/comments/ReplyList";
import {
    fetchRepliesAction,
    isLoggedIn
} from "../../store/actions/async/CommentActions";
import CommentLoading from "../../components/comments/CommentLoading";
import {SUCCEEDED} from "../../store/actions/async";
import CommentError from "../../components/comments/CommentError";
import NewReplyContainer from "./NewReplyContainer";

class CommentReplyListContainer extends Component {
    componentDidMount() {
        this.fetchReplyData();
    }

    fetchReplyData() {
        const {pk, fetchReplies} = this.props;
        fetchReplies(pk);
    }

    render() {
        const {replies, loading, status, pk} = this.props;
        let commentRepliesContentToRender;
        if (loading) {
            commentRepliesContentToRender = (
                <CommentLoading />
            );
        } else {
            if (status === SUCCEEDED) {
                commentRepliesContentToRender =
                    <ReplyList commentReplies={replies} />;
            } else {
                commentRepliesContentToRender = (
                    <CommentError
                        retry={() => this.fetchReplyData()}
                        message="Could not load Replies" />
                );
            }
        }

        return (
            <Comment.Group size="small">
                <Header
                    className="comment-teal" as="h4"
                    dividing>
                    Replies
                </Header>
                {isLoggedIn(localStorage) ? (
                    <NewReplyContainer
                        placeholder="Reply to this comment..."
                        slug={pk} />
                ): (
                    <div className="center">
                        <Link to="/login">
                            <Button size="tiny" color="teal">
                                Login to have your say
                            </Button>
                        </Link>
                    </div>
                )}
                {commentRepliesContentToRender}
            </Comment.Group>
        );
    }
}

CommentReplyListContainer.propTypes = {
    replies: PropTypes.array.isRequired,
    loading: PropTypes.bool.isRequired,
    status: PropTypes.string.isRequired,
    pk: PropTypes.number.isRequired,
    fetchReplies: PropTypes.func.isRequired
};

const mapStateToProps = (state, props) => {
    const reply = state.replies[props.pk];
    // if you closely examine the way we are handling state for the replies,
    // you notice that we can't use a single list of replies
    // since multiple comment's each with their own replies can appear in a single
    // component. This state cannot be given a default state in the reducers though
    // since we do not know which replies we need until a user tries to fetch them
    // The goal here is therefore to handle the situation where the component tries to render
    // before the commentReply state has been instantiated
    const blank = typeof reply === 'undefined' || !reply;
    return {
        replies: blank ? [] : reply.replies ? reply.replies : [],
        loading: blank ? true : !!reply.loading,
        status: blank ? SUCCEEDED : reply.status ? reply.status : SUCCEEDED
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        fetchReplies: (pk) => dispatch(fetchRepliesAction(pk))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(CommentReplyListContainer);
