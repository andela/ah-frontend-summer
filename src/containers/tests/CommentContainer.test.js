import {render, shallow} from "enzyme";
import React from "react";
import {applyMiddleware, createStore} from "redux";
import thunk from "redux-thunk";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import data from '../../components/__tests__/Comments.test';
import CommentBodyContainer from "../Comments/CommentBodyContainer";
import {SUCCEEDED} from "../../store/actions/async";
import CommentContainer from "../Comments/CommentContainer";
import CommentListContainer from "../Comments/CommentListContainer";
import ReplyContainer from "../Comments/ReplyContainer";
import CommentReplyListContainer from "../Comments/CommentReplyListContainer";
import {NewCommentContainer} from "../Comments/NewCommentContainer";
import NewReplyContainer from "../Comments/NewReplyContainer";
import reducer from "../../store/reducers";

const mockStore = createStore(reducer, applyMiddleware(thunk));

describe('CommentBodyContainer Tests', () => {
    it('renders without crashing', () => {
        render(<CommentBodyContainer
            body={data.comment.body}
            editComment={() => {
            }}
            editing={false}
            finishEditing={() => {
            }}
            loading={false}
            pk={data.comment.id}
            status={SUCCEEDED} />);
    });

    describe('instance tests', () => {
        let props, instance;
        beforeEach(() => {
            props = {
                body: "body",
                editComment: jest.fn(),
                editing: false,
                finishEditing: jest.fn(),
                loading: false,
                pk: 2,
                status: SUCCEEDED
            };
            const wrapper = shallow(<CommentBodyContainer {...props} />);
            instance = wrapper.instance();
        });

        it('componentDidUpdate calls finishEditing properly', () => {
            instance.state.loading = false;
            instance.state.status = SUCCEEDED;
            instance.componentDidUpdate({loading: true});
            expect(props.finishEditing).toHaveBeenCalled();
        });

        it('render returns expected content', () => {
            expect(instance.render()).toBeTruthy();
            instance.editing = true;
            expect(instance.render()).toBeTruthy();
        });

        it('updateComment correctly calls editComment', () => {
            instance.state.body = "updatedBody";
            instance.updateComment();
            expect(props.editComment).toHaveBeenCalledWith(props.pk, "updatedBody");
        });

        it('reset body resets the body in the state', () => {
            expect(instance.state.body).toEqual(props.body);
            instance.state.body = "updatedBody";
            expect(instance.state.body).toEqual("updatedBody");
            instance.resetBody();
            expect(instance.state.body).toEqual(props.body);
        });


    });
});

describe('CommentContainer Tests', () => {
    it('renders without crashing', () => {
        shallow(<CommentContainer
            editComment={() => {
            }} comment={data.comment} deleteLoading={false} doDeleteComment={() => {
            }} />);
    });

    describe('mapStateToProps and mapDispatchToProps', () => {
        it('returns the right props', () => {
            const {mapStateToProps, mapDispatchToProps} = require("../Comments/CommentContainer");
            expect(mapStateToProps({
                comments: {
                    deleteComment: {loading: false},
                    comments: [{id: 2, loading: false, status: SUCCEEDED}]
                }
            }, {comment: {id: 2}})).toEqual({
                loading: false,
                status: SUCCEEDED,
                deleteLoading: false
            });
            let props = mapDispatchToProps();
            expect(props).toHaveProperty('editComment');
            expect(props).toHaveProperty('doDeleteComment');
        });
    });
});

describe('CommentListContainer Tests', () => {
    it('renders without crashing', () => {
        shallow(<CommentListContainer
            loading={false} status={SUCCEEDED}
            comments={[data.comment]}
            fetchComments={() => {
            }} slug="slug" />);
        render(<Provider store={mockStore}>
            <BrowserRouter>
                <CommentListContainer
                    fetchComments={() => {
                    }} slug="slug" />
            </BrowserRouter>
               </Provider>);
    });
});

describe('CommentReplyListContainer Tests', () => {
    it('renders without crashing', () => {
        shallow(<CommentReplyListContainer
            status={SUCCEEDED} loading={false}
            fetchReplies={() => {
            }} pk={data.comment.id}
            replies={[]} />);
    });
});

describe('NewCommentContainer Tests', () => {
    it('renders without crashing', () => {
        shallow(<NewCommentContainer
            loading={false} createComment={() => {
            }} status={SUCCEEDED} slug="slug" />);
    });

    describe('mapStateToProps and mapDispatchToProps', () => {
        it('returns the right props', () => {
            const {mapStateToProps, mapDispatchToProps} = require("../Comments/NewCommentContainer");
            expect(mapStateToProps({
                comments: {
                    newComment: {
                        loading: false,
                        status: SUCCEEDED
                    }
                }
            })).toEqual({
                loading: false,
                status: SUCCEEDED,
            });
            let props = mapDispatchToProps();
            expect(props).toHaveProperty('createComment');
        });
    });

    describe('instance tests', () => {
        let props, instance;

        beforeEach(() => {
            props = {
                createComment: jest.fn(),
                loading: false,
                slug: "slug",
                status: SUCCEEDED,
                placeholder: "placeholder",
            };

            const wrapper = shallow(<NewCommentContainer {...props} />);
            instance = wrapper.instance();
        });

        it('componentDidUpdate calls clearBody properly', () => {
            instance.state.loading = false;
            instance.state.status = SUCCEEDED;
            instance.clearBody = jest.fn();
            instance.componentDidUpdate({loading: true});
            expect(instance.clearBody).toHaveBeenCalled();
        });

        it('createNewComment correctly calls createComment', () => {
            instance.state.body = "updatedBody";
            instance.createNewComment();
            expect(props.createComment).toHaveBeenCalledWith(props.slug, "updatedBody");
        });

        it('clearBody clears the body', () => {
            instance.clearBody();
            expect(instance.state.body).toEqual("");
        });


    });
});

describe('NewReplyContainer Tests', () => {
    it('renders without crashing', () => {
        shallow(<NewReplyContainer />);
    });
});

describe('Reply Container Tests', () => {
    it('renders without crashing', () => {
        shallow(<ReplyContainer
            deleteLoading={false} reply={data.reply}
            doDeleteReply={() => {
            }} doEditReply={() => {
            }} />);
    });
});
