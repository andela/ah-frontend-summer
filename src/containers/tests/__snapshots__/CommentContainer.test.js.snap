// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeDislikeComments tests renders consistently 1`] = `
<Fragment>
  <CommentAction
    as="a"
  >
    <Icon
      as="i"
      color={null}
      name="thumbs up"
      onClick={[Function]}
      size="large"
    />
  </CommentAction>
  <Label
    basic={true}
    color="grey"
    pointing="left"
    size="tiny"
  >
    3
  </Label>
  <CommentAction
    as="a"
  />
  <CommentAction
    as="a"
  />
  <CommentAction
    as="a"
  />
  <CommentAction
    as="a"
  >
    <Icon
      as="i"
      color={null}
      name="thumbs down"
      onClick={[Function]}
      size="large"
    />
  </CommentAction>
  <Label
    basic={true}
    color="grey"
    pointing="left"
    size="tiny"
  >
    1
  </Label>
  <CommentAction
    as="a"
  />
</Fragment>
`;
